tests/aggregation_regress/tests.py

```diff
@@ -360,6 +361,40 @@ def test_aggrate_annotation(self):
             {'c__max': 3}
         )
 
+    def test_conditional_aggregate_of_subquery(self):
+        """
+        Regression test for #25307: AttributeError: 'WhereNode' object has no
+        attribute 'get_source_expressions'.
+
+        Tests conditional aggregations over subqueries.
+        """
+        # Conditional aggregation on a limited queryset.
```

Comment:
Ahh `group_by` is [trie-valued](https://github.com/django/django/blob/master/django/db/models/sql/query.py#L157-L163). I think `isinstance(self.group_by, list)` as something to do with using `values()`.